src/Microsoft.ML.Core/Utilities/NormStr.cs

```diff
@@ -136,6 +136,14 @@ public NormStr Get(ReadOnlyMemory<char> str, bool add = false)
                 }
                 Contracts.Assert(ins == -1);
 
+                if(createNewStr)
+                {
+                    // To avoid the case where 'str' actually stores a string with the
+                    // content of a whole row in the dataset, a new 'str' is created
+                    // See issue #4571 and PR #4576
```

Comment:
The shortlink issue/PR number won't survive repo forks, and the unrelenting march of time. You may want full links.
```suggestion
                    // See issue https://github.com/dotnet/machinelearning/issues/4571 
                    // and PR https://github.com/dotnet/machinelearning/pull/4576
```